src/main/java/org/jboss/remoting3/remote/ClientConnectionOpenListener.java

```diff
@@ -401,7 +401,7 @@ public void handleEvent(final ConduitStreamSourceChannel channel) {
                         factoryOperator = and(ClientConnectionOpenListener.this.saslClientFactoryOperator, factoryOperator);
                         final SaslClient saslClient;
                         try {
-                            saslClient = configurationClient.createSaslClient(uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
+                            saslClient = createSaslClient(configurationClient, uri, ClientConnectionOpenListener.this.configuration, serverSaslMechs, factoryOperator);
```

Comment:
Instead of doing this, it should wrap the SASL client in a PriviliegedSaslClient.